test/extended/setup.sh

```diff
@@ -4,219 +4,238 @@
-    # put change there - only want this for extended tests
-    echo "[INFO] Turn on audit logging"
-    cp ${SERVER_CONFIG_DIR}/master/master-config.yaml ${SERVER_CONFIG_DIR}/master/master-config.orig2.yaml
-    openshift ex config patch ${SERVER_CONFIG_DIR}/master/master-config.orig2.yaml --patch="{\"auditConfig\": {\"enabled\": true}}"  > ${SERVER_CONFIG_DIR}/master/master-config.yaml
-
-    # Similar to above check, if the XFS volume dir mount point exists enable
-    # local storage quota in node-config.yaml so these tests can pass:
-    if [ -d "/mnt/openshift-xfs-vol-dir" ]; then
-	# The ec2 images usually have ~5Gi of space defined for the xfs vol for the registry; want to give /registry a good chunk of that
-	# to store the images created when the extended tests run
-      sed -i 's/perFSGroup: null/perFSGroup: 4480Mi/' $NODE_CONFIG_DIR/node-config.yaml
-    fi
-    echo "[INFO] Using VOLUME_DIR=${VOLUME_DIR}"
-
-    # This is a bit hacky, but set the pod gc threshold appropriately for the garbage_collector test.
-    os::util::sed 's/\(controllerArguments:\ \)null/\1\n    terminated-pod-gc-threshold: ["100"]/' \
-      ${MASTER_CONFIG_DIR}/master-config.yaml
-
-    start_os_server
-
-    export KUBECONFIG="${ADMIN_KUBECONFIG}"
-
-    install_registry
-    wait_for_registry
-    DROP_SYN_DURING_RESTART=1 CREATE_ROUTER_CERT=1 install_router
-
-    echo "[INFO] Creating image streams"
-    oc create -n openshift -f examples/image-streams/image-streams-centos7.json --config="${ADMIN_KUBECONFIG}"
-  else
-    # be sure to set VOLUME_DIR if you are running with TEST_ONLY
-    echo "[INFO] Not starting server, VOLUME_DIR=${VOLUME_DIR:-}"
-  fi
+#		and then tests are executed.	Tests that depend on fine grained setup should
+#		be done in other contexts.
+function os::test::extended::setup () {
+	# build binaries
+	if [[ -z "$(os::build::find-binary ginkgo)" ]]; then
+		hack/build-go.sh vendor/github.com/onsi/ginkgo/ginkgo
+	fi
+	if [[ -z "$(os::build::find-binary extended.test)" ]]; then
+		hack/build-go.sh test/extended/extended.test
+	fi
+	if [[ -z "$(os::build::find-binary openshift)" ]]; then
+		hack/build-go.sh
+	fi
+
+	os::util::environment::setup_time_vars
+
+	# ensure proper relative directories are set
+	export GINKGO="$(os::build::find-binary ginkgo)"
+	export EXTENDEDTEST="$(os::build::find-binary extended.test)"
+	export EXTENDED_TEST_PATH="${OS_ROOT}/test/extended"
+	export KUBE_REPO_ROOT="${OS_ROOT}/vendor/k8s.io/kubernetes"
+
+	# output tests instead of running
+	if [[ -n "${SHOW_ALL:-}" ]]; then
+		full_test_list="$( "${EXTENDEDTEST}" --ginkgo.dryRun --ginkgo.noColor )"
+		selected_tests=()
+		for test in "${full_test_list[@]}"; do
+			if grep -q "ok" <<< "${test}" && grep -qv "skip" <<< "${test}"; then
+				selected_tests+=( "${test}" )
+			fi
+		done
+		if [[ "${#selected_tests[@]}" -eq 0 ]]; then
+			os::log::error "The provided focus did not select any tests!"
+			exit 1
+		fi
+		sort <<< "${selected_tests[@]}"
+		exit 0
```

Comment:
a return will result in the calling script proceeding, which will try to execute tests while the server hasn't been started.